File: dotnet-core/Tea/Decorators/Elements/ElementDecorator.cs
using OpenQA.Selenium;

namespace Tea
{
    public class ElementDecorator : Element
    {
        protected readonly Element Element;

        public ElementDecorator(Element element)
        {
            Element = element;
        }
        public override By By => Element?.By;

        public override string Text => Element?.Text;

        public override bool? Enabled => Element?.Enabled;

        public override bool? Displayed => Element?.Displayed;

        public override void Click()
        {
            Element?.Click();
        }

        public override string GetAttribute(string attributeName)
        {
            return Element?.GetAttribute(attributeName);
        }

        public override void TypeText(string text)
        {
           Element?.TypeText(text);
        }
    }
}

File: java/tea-framework/src/main/java/com/tea/framework/SeleniumWrapper/Elements/ElementDecorator.java
package com.tea.framework.SeleniumWrapper.Elements;

public class ElementDecorator extends Element {
    protected Element element;

    public ElementDecorator(Element element) {
        this.element = element;
        this.text = element.getText();
        this.by = element.by;
        this.enabled = element.isEnabled();
        this.displayed = element.isDisplayed();
    }

    @Override
    public boolean isEnabled() {
        return element.isEnabled();
    }

    @Override
    public boolean isDisplayed() {
        return element.isDisplayed();
    }

    @Override
    public void Click() {
        element.Click();
    }

    @Override
    public String GetAttribute(String attributeName) {
        return element.GetAttribute(attributeName);
    }

    @Override
    public void TypeText(String text) {
        element.TypeText(text);
    }
}